### PR TITLE
Adding token refresher to EKS-A release bundle

### PR DESCRIPTION
**Pull request by jonahjon**
Signed-off-by: jonahjon <jonahjones094@gmail.com>

Current tests are failing  since the new image isn't being pushed to Public ECR. This is adding it to the bundle.

### CODE REVIEW

File: release/api/v1alpha1/bundle_types.go [hunk 1]
```diff
@@ -235,6 +235,7 @@ type FluxBundle struct {
 type PackageBundle struct {
 	Version    string `json:"version,omitempty"`
 	Controller Image  `json:"packageController"`
+	Refreshser Image  `json:"tokenRefresher"`
```
Comment (abhay-krishna) [inline release/api/v1alpha1/bundle_types.go:238]: typo
```suggestion
	Refresher Image  `json:"tokenRefresher"`
```

Comment (abhay-krishna) [reply release/api/v1alpha1/bundle_types.go:238]: I would actually prefer to use `TokenRefresher` as the field name to make intentions clear.

**Comment by TerryHowe** [comment]
conflicts!

**Comment by TerryHowe** [comment]
I resolved the conflicts

File: config/crd/bases/anywhere.eks.amazonaws.com_cloudstackdatacenterconfigs.yaml [hunk 2]
```diff
@@ -39,12 +39,13 @@ spec:
               of CloudStackDatacenterConfig
             properties:
               account:
```
Comment (TerryHowe) [inline config/crd/bases/anywhere.eks.amazonaws.com_cloudstackdatacenterconfigs.yaml:41]: This change looks like it shouldn't be here

File: config/manifest/eksa-components.yaml [hunk 3]
```diff
@@ -3056,12 +3086,13 @@ spec:
               of CloudStackDatacenterConfig
             properties:
               account:
```
Comment (TerryHowe) [inline config/manifest/eksa-components.yaml:3088]: and these changes

File: config/crd/bases/anywhere.eks.amazonaws.com_cloudstackdatacenterconfigs.yaml (hunk 2 above)
Comment (TerryHowe) [inline config/crd/bases/anywhere.eks.amazonaws.com_cloudstackdatacenterconfigs.yaml:41]: It would be nice to get rid of these cloudstack changes.

**Comment by TerryHowe** [comment]
tests borked

**Comment by jonahjon** [comment]
Looks to be an issue due to a recent cloudtstack PR https://github.com/aws/eks-anywhere-build-tooling/pull/1170

File: release/pkg/bundles/package-controller.go [hunk 4]
```diff
@@ -23,42 +23,50 @@ import (
 	anywherev1alpha1 "github.com/aws/eks-anywhere/release/api/v1alpha1"
 	"github.com/aws/eks-anywhere/release/pkg/constants"
 	releasetypes "github.com/aws/eks-anywhere/release/pkg/types"
+	bundleutils "github.com/aws/eks-anywhere/release/pkg/util/bundles"
 	"github.com/aws/eks-anywhere/release/pkg/version"
 )
 
 func GetPackagesBundle(r *releasetypes.ReleaseConfig, imageDigests map[string]string) (anywherev1alpha1.PackageBundle, error) {
-	artifacts := r.BundleArtifactsTable["eks-anywhere-packages"]
+	//artifacts := r.BundleArtifactsTable["eks-anywhere-packages"]
```
Comment (a-cool-train) [inline release/pkg/bundles/package-controller.go:31]: Can we remove this from the code if it is not necessary?

**Review by a-cool-train — CHANGES_REQUESTED** [review]
@jonahjon, we need to also extract the ecr refresher cron job in case of download images and import images. This is to support the disconnected area.

**Comment by jonahjon** [comment]
> @jonahjon, we need to also extract the ecr refresher cron job in case of download images and import images. This is to support the disconnected area.

What does that mean?

**Comment by jonahjon** [comment]
> @jonahjon, we need to also extract the ecr refresher cron job in case of download images and import images. This is to support the disconnected area.

Also can we just do that in another PR since this one is blocking the release pipeline.

**Comment by jonahjon** [comment]
tide

**Comment by jonahjon** [comment]
Why prow decides to re-test 7 jobs that passed a few minutes ago is a mystery to me.

**Pull request merged**